[BUG] TypeScript's `baseUrl` is not honoured

<!-- ⚠️⚠️ Do not delete this template ⚠️⚠️ -->

<!-- 🔎 Search existing issues to avoid creating duplicates. -->
<!-- 🧪 Test using the latest Playwright release to see if your issue has already been fixed -->
<!-- 💡 Provide enough information for us to be able to reproduce your issue locally -->

### System info
- Playwright Version: 1.32.3
- Operating System: macOS 13.3.1
- Browser: N/A
- Other info:

### Source code

- [x] I provided exact source code that allows reproducing the issue locally.

<!-- For simple cases, please provide a self-contained test file along with the config file -->
<!-- For larger cases, you can provide a GitHub repo you created for this issue -->
<!-- If we can not reproduce the problem locally, we won't be able to act on it -->
<!-- You can still file without the exact code and we will try to help, but if we can't repro, it will be closed -->

**Link to the GitHub repository with the repro**

https://github.com/OliverJAsh/playwright-ts-paths-issues/tree/base-url-ignored

**Test file (self-contained)**

```js
import * as Foo from "Foo";
console.log(Foo);
```

`tsconfig.json`:
```json
    "baseUrl": ".",
    "paths": {
      "shared/*": ["./shared/*"]
    }
```

**Steps**
```console
$ yarn
$ rm -rf .cache && tsc
$ rm -rf .cache && playwright test
```

**Expected**

`playwright test` should resolve all modules correctly, as per the behaviour of `tsc`.

**Actual**

`playwright test` results in the following error:

```
Error: Cannot find module 'Foo'
Require stack:
- /Users/oliverash/Development/playwright-ts-paths-issues/tests/example.spec.ts
```

**Workaround**

```diff
diff --git a/tsconfig.json b/tsconfig.json
index 0eaedd8..89c2a6f 100644
--- a/tsconfig.json
+++ b/tsconfig.json
@@ -5,6 +5,7 @@
     "rootDir": ".",
     "baseUrl": ".",
     "paths": {
+      "*": ["./*"],
       "shared/*": ["./shared/*"]
     }
   },

```

This shouldn't be necessary because Playwright should use the `baseUrl` to resolve the import specifier.

--- a/packages/playwright-test/src/common/transform.ts
+++ b/packages/playwright-test/src/common/transform.ts
@@ -38,11 +38,11 @@ function validateTsConfig(tsconfig: TsConfigLoaderResult): ParsedTsConfigData | 
     return;
   // Make 'baseUrl' absolute, because it is relative to the tsconfig.json, not to cwd.
   const absoluteBaseUrl = path.resolve(path.dirname(tsconfig.tsConfigPath), tsconfig.baseUrl);
-  const paths = tsconfig.paths || { '*': ['*'] };
+  const pathsFallback = [{ key: '*', values: ['*'] }];
   return {
     allowJs: tsconfig.allowJs,
     absoluteBaseUrl,
-    paths: Object.entries(paths).map(([key, values]) => ({ key, values }))
+    paths: Object.entries(tsconfig.paths || {}).map(([key, values]) => ({ key, values })).concat(pathsFallback)
   };
 }
 

--- a/tests/playwright-test/resolver.spec.ts
+++ b/tests/playwright-test/resolver.spec.ts
@@ -164,6 +164,43 @@ test('should respect baseurl w/o paths', async ({ runInlineTest }) => {
   expect(result.output).not.toContain(`Could not`);
 });
 
+test('should fallback to *:* when baseurl and paths are specified', async ({ runInlineTest }) => {
+  const result = await runInlineTest({
+    'foo/bar/util/b.ts': `
+      export const foo = 42;
+    `,
+    'shared/x.ts': `
+      export const x = 43;
+    `,
+    'dir2/tsconfig.json': `{
+      "compilerOptions": {
+        "target": "ES2019",
+        "module": "commonjs",
+        "lib": ["esnext", "dom", "DOM.Iterable"],
+        "baseUrl": "..",
+        "paths": {
+          "shared/*": ["./shared/*"],
+        },
+      },
+    }`,
+    'dir2/inner.spec.ts': `
+      // This import should pick up ../foo/bar/util/b due to baseUrl and *:* fallback.
+      import { foo } from 'foo/bar/util/b';
+      // This import should pick up ../shared/x due to baseUrl+paths.
+      import { x } from 'shared/x';
+      import { test, expect } from '@playwright/test';
+      test('test', ({}, testInfo) => {
+        expect(foo).toBe(42);
+        expect(x).toBe(43);
+      });
+    `,
+  });
+
+  expect(result.exitCode).toBe(0);
+  expect(result.passed).toBe(1);
+  expect(result.output).not.toContain(`Could not`);
+});
+
 test('should respect complex path resolver', async ({ runInlineTest }) => {
   const result = await runInlineTest({
     'playwright.config.ts': `

